internal/integration/oidc.go

```diff
@@ -227,17 +228,25 @@ func GetRequest(url string, headers map[string]string) (*http.Request, error) {
 }
 
 func CheckRedirect(req *http.Request) (*url.URL, error) {
-	client := &http.Client{
+	c := &http.Client{
 		CheckRedirect: func(req *http.Request, via []*http.Request) error {
 			return http.ErrUseLastResponse
 		},
 	}
-	resp, err := client.Do(req)
+	resp, err := c.Do(req)
 	if err != nil {
 		return nil, err
 	}
 	defer resp.Body.Close()
 
+	if resp.StatusCode != 302 {
+		body, err := io.ReadAll(resp.Body)
+		if err != nil {
+			return nil, err
+		}
+		return nil, fmt.Errorf("bad request: %s", string(body))
+	}
+
```

Comment:
Was helpful when something went wrong to see why, otherwise, the `Location` header will no be there, but difficult to see what happened without breakpoints